### DISCUSSION TITLE
How do you locally build QMK Firmware for the rollow keyboard?

### DISCUSSION THREAD

**Discussion by alexchaveztech** [manna-harbour/miryoku#204] [Q&A] [answered]
What are the steps to build locally the QMK firmware for the rollow keyboard using Miryoku. I am able to do it for crkbd but struggling with the rollow. Thank in advanced!

**Comment by manna-harbour** [comment, marked as answer]
Rollow isn't in QMK master, so I won't add rollow support to the `miryoku` branch yet. But here it is in a temporary branch: https://github.com/manna-harbour/miryoku_qmk/tree/rollow-test.

You need `keyboards/rollow/` from https://github.com/barbellboards/Rollow-QMK/tree/barbellboards/rollow and `keyboards/rollow/keymaps/manna-harbour_miryoku/` from https://github.com/manna-harbour/miryoku_qmk/tree/rollow-test.

If you're building locally you can copy, cherry pick, or merge, etc. Also see https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#local-builds.

Ordinarily you can also use the Miryoku QMK build workflows to automatically merge the necessary branches at build time, but the `rollow-test` branch is now too old for that and needs rebasing.

Encoder clicks are the tertiary thumb keys, but thumb combos are also enabled: https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#thumb-combos. Support for customising encoder rotations will be added (https://github.com/manna-harbour/miryoku/issues/94) but for now you can do it according to https://github.com/qmk/qmk_firmware/blob/master/docs/feature_encoders.md#encoder-map-idencoder-map in `keyboards/rollow/keymaps/manna-harbour_miryoku/keymap.c`.

**Comment by alexchaveztech** [comment]
Thank you so much for your quick response. I followed your instructions and was able to compile the firmware successfully including the thumb keys shortcuts. I am still testing the encoders, but that function is not important for me at this point.
